pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -416,22 +416,23 @@ public void destroy() {
 
     // If offHeapAllocation, then gather some statistics before  destroying the segment.
     if (isOffHeapAllocation) {
-      RealtimeSegmentStatsHistory.SegmentStats segmentStats = new RealtimeSegmentStatsHistory.SegmentStats();
-      for (Map.Entry<String, MutableDictionary> entry : dictionaryMap.entrySet()) {
-        RealtimeSegmentStatsHistory.ColumnStats columnStats = new RealtimeSegmentStatsHistory.ColumnStats();
-        columnStats.setCardinality(entry.getValue().length());
-        columnStats.setAvgColumnSize(entry.getValue().getAvgValueSize());
-        segmentStats.setColumnStats(entry.getKey(), columnStats);
+      if (numDocsIndexed > 0) {
```

Comment:
Can the two if's be combined into one, or is there logic inside the outer 'if' that is not inside the inner 'if'?

Reply:
I would prefer to leave them as separate if statements instead of relying on order of execution of the two conditions separated by &&